exercises/bracket-push/example.rs

```diff
@@ -0,0 +1,62 @@
+// The code below is a stub. Just enough to satisfy the compiler.
+// In order to pass the tests you can add-to or change any of this code.
+
+use std::collections::HashMap;
+pub struct Brackets {
+    all: Vec<char>,
+    pairs: HashMap<char, char>,
+}
+
+impl<'a> From<&'a str> for Brackets {
+    fn from(i: &str) -> Self {
+        Brackets::new(String::from(i), None)
+    }
+}
+
+impl Brackets {
+    pub fn new(s: String, pairs: Option<HashMap<char, char>>) -> Self {
+        let p = match pairs {
+            Some(x) => x,
+            None => Brackets::default(),
+        };
+
+        let valid = p.keys().chain(p.values()).collect::<Vec<_>>();
+        let all = s.chars().filter(|c| valid.contains(&c)).collect::<Vec<char>>();
+
+        Brackets {
+            all: all,
+            pairs: p.clone(),
+        }
+    }
+
+    fn default() -> HashMap<char, char> {
+        let mut d = HashMap::new();
+        d.insert('[', ']');
+        d.insert('{', '}');
+        d.insert('(', ')');
+        d
+    }
+
+    pub fn are_balanced(&self) -> bool {
+        let mut unmatched: Vec<char> = Vec::new();
+
+        for b in self.all.clone() {
+            if unmatched.is_empty() {
+                unmatched.push(b);
+            } else {
+                let t = unmatched.pop().unwrap();
+
+                if let Some(c) = self.pairs.get(&t) {
+                    if *c != b {
+                        unmatched.push(t);
```

Comment:
Probably [`last`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.last)

You are right that I meant to look at the last element without popping it. I wasn't specific at the time since I hadn't examined whether it was possible, now I can be specific about `last`